core/Enclave/key_operation.cpp

```diff
@@ -387,8 +387,8 @@ sgx_status_t ehsm_aes_gcm_decrypt(ehsm_data_t *aad,
                           block_mode,
                           cipherblob->data,
                           cipherblob->datalen - EH_AES_GCM_IV_SIZE - EH_AES_GCM_MAC_SIZE,
-                          aad->data,
-                          aad->datalen,
+                          aad ? aad->data : NULL,
+                          aad ? aad->datalen : 0,
                           iv,
```

Comment:
https://github.com/intel/ehsm/blob/main/core/Enclave/key_operation.cpp#L199
Is there any similar bug in aes_gcm_encrypt() ?

Reply:
`aes_gcm_encrypt()` only accesses `aad` when `aad` is not null.

Reference: https://github.com/intel/ehsm/blob/main/core/Enclave/openssl_operation.cpp#L94

Reply:
Sorry. I am talking about ehsm_aes_gcm_encrypt().  I mean if aad is null, will this [line ](https://github.com/intel/ehsm/blob/main/core/Enclave/key_operation.cpp#L199) in ehsm_aes_gcm_encrypt() also cause to null dereference?

Reply:
Yes, `ehsm_aes_gcm_encrypt` has the same problem. Fixed in latest commit.